### PR TITLE
feat: add secret types.

### PR DESCRIPTION
**Pull request by Atheuz**
## Change Summary

Added SecretStr and SecretBytes.

## Related issue number

#443 

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `HISTORY.rst` has been updated

### CODE REVIEW

**Comment by samuelcolvin** [comment]
Looking good.

Please change `self.value` to `self._secret_value` so it's private.

Please add a bit too docs somewhere explaining the purpose and how to use these types.

File: tests/test_schema.py [hunk 1]
```diff
@@ -517,6 +519,38 @@ class Model(BaseModel):
...
+        'properties': {'a': {'title': 'A', 'type': 'string'}},
+        'required': ['a'],
+    }
+    base_schema['properties']['a']['format'] = expected_schema
+
+    assert Model.schema() == base_schema
+
+
+@pytest.mark.parametrize('field_type,expected_schema', [(SecretBytes, 'secret-bytes')])
+def test_secret_bytrs_type(field_type, expected_schema):
```
Comment (LasseGravesen) [inline tests/test_schema.py]: typo

**Comment by Atheuz** [comment]
Anymore comments @samuelcolvin?

**Comment by samuelcolvin** [comment]
I think you've messed up your branch and there are lots of changes on your PR that shouldn't be there.

You'll need to resolve that. You might be best to reset soft to upstream master and create a new PR.

**Comment by LasseGravesen** [comment]
@samuelcolvin Ugh. I saw that a branch had been merged and tried to rebase the changes into my branch from upstream, but that must have messed it up. I'll make a new PR later today.

**Pull request closed without merging**